netdata/netdata
# eBPF Shared Memory system call tracking

##### Summary

This PR brings eBPF-based shared memory system call tracking. https://github.com/netdata/product/issues/2159

##### Component Name

ebpf.plugin

##### Test Plan

- Compile this branch.
- Check that these charts shows up:
  - `system.shared_memory_calls`
  - `apps.shmget_call`
  - `apps.shmat_call`
  - `apps.shmdt_call`
  - `apps.shmctl_call`
- Run some apps/programs that use shared memory and see charts update.

Tested with:

- Ubuntu 21.04 (Linux 5.11.0-31)
- Ubuntu 18.04 (Linux 4.15)
- CentOS 7 (Linux 3.10)
